llvm/lib/Target/RISCV/RISCVInsertVSETVLI.cpp

```diff
@@ -48,6 +50,44 @@ static cl::opt<bool> UseStrictAsserts(
 
 namespace {
 
+// For the SSA form, we could just use the getVRegDef to take Reaching
+// definition. For the non-SSA, we retrieve reaching definition for specific
+// register from LiveInterval/VNInfo.
+template <typename T>
+static T *getReachingDefMI(Register Reg, T *MI, const MachineRegisterInfo *MRI,
+                           const LiveIntervals *LIS) {
+  if (MRI->isSSA())
+    return MRI->getVRegDef(Reg);
+
+  if (!MI)
+    return MRI->getUniqueVRegDef(Reg);
```

Comment:
You can use `getUniqueVRegDef` for SSA too. So we can merge these two conditions.

```
if (MRI->isSSA() || !MI)
  return MRI->getUniqueVRegDef(Reg);
```

Reply:
Done